integrations/opensearch/src/haystack_integrations/components/retrievers/open_search_hybrid_retriever.py

```diff
@@ -0,0 +1,322 @@
+# SPDX-FileCopyrightText: 2023-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Dict, List, Literal, Optional, Union
+
+from haystack import Pipeline, default_from_dict, default_to_dict, logging
+from haystack.components.embedders import SentenceTransformersTextEmbedder
+from haystack.components.joiners import DocumentJoiner
+from haystack.components.joiners.document_joiner import JoinMode
+from haystack.document_stores.types import FilterPolicy
+from haystack.lazy_imports import LazyImport
+from haystack.utils import ComponentDevice
+
+from haystack_integrations.components.retrievers.opensearch import OpenSearchBM25Retriever, OpenSearchEmbeddingRetriever
+from haystack_integrations.document_stores.opensearch import OpenSearchDocumentStore
+
+logger = logging.getLogger(__name__)
+
+# Use LazyImport to conditionally import haystack super_component, which is only available in newer versions
+with LazyImport(
+    "To use the OpenSearchHybridRetriever you need more recent version of haystack. Run 'pip install haystack-ai>=2.13.0'"  # noqa: E501
+) as haystack_imports:
+    from haystack import super_component
+
+# Trigger an error message when the OpenSearchHybridRetriever is imported without haystack-ai>=2.13.0
+haystack_imports.check()
+
+
+@super_component
+class OpenSearchHybridRetriever:
```

Comment:
I assume we can't add OpenSearchHybridRetriever to the init file where the other two retrievers are also listed because then users who import OpenSearchBM25Retriever also are required to have haystack>=2.13 installed? https://github.com/deepset-ai/haystack-core-integrations/blob/main/integrations/opensearch/src/haystack_integrations/components/retrievers/opensearch/__init__.py